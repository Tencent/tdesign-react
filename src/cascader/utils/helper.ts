import { TreeNode, CascaderContextType, TreeNodeValue, TreeOptionData } from '../interface';

/**
 * 获取node的全部路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}

/**
 * treeValue计算方法
 * @param value
 * @returns
 */
export const getTreeValue = (value: CascaderContextType['value']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(value)) {
    if (value.length > 0 && typeof value[0] === 'object') {
      treeValue = (value as TreeOptionData[]).map((val) => val.value);
    } else if (value.length) {
      treeValue = value as TreeNodeValue[];
    }
  } else if (value) {
    if (typeof value === 'object') {
      treeValue = [(value as TreeOptionData).value];
    } else {
      treeValue = [value];
    }
  }
  return treeValue;
};

/**
 * 判断node是否在页面上
 * @param node Node
 */
export function isInPage(node: Node) {
  return node === document.body ? false : document.body.contains(node);
}

export default {
  getFullPathLabel,
};
