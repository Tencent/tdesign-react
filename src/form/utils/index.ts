import type { NamePath } from '../type';

// 获取 formMap 管理的数据
export function getMapValue(name: NamePath, formMapRef: React.MutableRefObject<Map<any, any>>) {
  if (!formMapRef.current) return;

  // 提取所有 map key
  const mapKeys = [...formMapRef.current.keys()];
  // 转译为字符串后比对 key 兼容数组格式
  const key = mapKeys.find((key) => String(key) === String(name));
  // 拿到 key 引用地址获取 value
  return formMapRef.current.get(key);
}

// 将数据整理成对象，数组 name 转化嵌套对象: ['user', 'name'] => { user: { name: '' } }
export function calcFieldValue(name: NamePath, value: any) {
  if (Array.isArray(name)) {
    const fieldValue = name.reduceRight((prev, curr) => {
      const arr = [];
      if (/^\d+$/.test(String(curr))) arr[curr] = prev;
      return arr.length ? arr : { [curr]: prev };
    }, value);
    return { ...fieldValue };
  }

  return { [name]: value };
}

// 通过对象数据类型获取 map 引用: { user: { name: '' } } => formMap.get(['user', 'name'])
export function travelMapFromObject(
  obj: Record<any, any>,
  formMapRef: React.MutableRefObject<Map<any, any>>,
  callback: Function,
) {
  for (const [mapName, formItemRef] of formMapRef.current.entries()) {
    // 支持嵌套数据结构
    if (Array.isArray(mapName)) {
      // 创建唯一临时变量 symbol
      const symbol = Symbol('name');
      let fieldValue = null;

      for (let i = 0; i < mapName.length; i++) {
        const item = mapName[i];
        if (Reflect.has(fieldValue || obj, item)) {
          fieldValue = Reflect.get(fieldValue || obj, item);
        } else {
          // 当反射无法获取到值则重置为 symbol
          fieldValue = symbol;
          break;
        }
      }

      // 非 symbol 说明获取到了值
      if (fieldValue !== symbol) {
        callback(formItemRef, fieldValue);
      }
    } else if (Reflect.has(obj, mapName)) {
      callback(formItemRef, obj[mapName]);
    }
  }
}
