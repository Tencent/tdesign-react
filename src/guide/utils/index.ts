import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
