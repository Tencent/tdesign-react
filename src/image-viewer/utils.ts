export const downloadFile = function (imgSrc) {
  const image = new Image();
  const name = imgSrc?.split?.('/').pop() || Math.random().toString(32).slice(2);

  image.setAttribute('crossOrigin', 'anonymous');

  image.onload = () => {
    const canvas = document.createElement('canvas');
    canvas.width = image.width;
    canvas.height = image.height;

    const context = canvas.getContext('2d');
    context.drawImage(image, 0, 0, image.width, image.height);
    canvas.toBlob((blob) => {
      const url = URL.createObjectURL(blob);
      const a = document.createElement('a');
      a.download = name;
      a.href = url;
      a.click();
      a.remove();
      URL.revokeObjectURL(url);
    });
  };
  image.src = imgSrc;
};
