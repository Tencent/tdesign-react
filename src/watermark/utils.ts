const toLowercaseSeparator = (key: string) => key.replace(/([A-Z])/g, '-$1').toLowerCase();

// style对象转字符串
export const getStyleStr = (style: React.CSSProperties): string =>
  Object.keys(style)
    .map((key: keyof React.CSSProperties) => `${toLowercaseSeparator(key)}: ${style[key]};`)
    .join(' ');
