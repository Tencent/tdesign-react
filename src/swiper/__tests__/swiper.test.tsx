import React from 'react';
import { render, fireEvent } from '@test/utils';
import Swiper from '../index';

const { SwiperItem } = Swiper;

describe('Swiper 组件测试', () => {
  // 测试渲染
  test('create', async () => {
    const { container } = render(<Swiper></Swiper>);
    expect(container.querySelectorAll('.t-swiper')).toHaveLength(1);
  });

  // 测试事件
  test('arrows event', async () => {
    const TestView = () => (
      <>
        <Swiper
          autoplay={false}
          loop
          navigation={{
            showSlideBtn: 'hover',
          }}
        >
          <SwiperItem>
            <div className="demo-item">1</div>
          </SwiperItem>
          <SwiperItem>
            <div className="demo-item">2</div>
          </SwiperItem>
          <SwiperItem>
            <div className="demo-item">3</div>
          </SwiperItem>
        </Swiper>
      </>
    );

    render(<TestView />);
    // 获取 element

    expect(document.querySelector('.t-swiper__container')).not.toBeNull();
    expect(document.querySelectorAll('.t-swiper__navigation-item').length).toBe(3);

    fireEvent.click(document.querySelectorAll('.t-swiper__navigation-item')[1]);
    fireEvent.mouseEnter(document.querySelectorAll('.t-swiper__navigation-item')[1]);
    fireEvent.mouseLeave(document.querySelectorAll('.t-swiper__navigation-item')[1]);
    expect(document.querySelectorAll('.t-swiper__navigation-item')[1].classList.contains('t-is-active')).toBeTruthy();
  });

  // 测试事件
  test('event', async () => {
    const TestView = () => (
      <>
        <Swiper loop>
          <SwiperItem>
            <div className="demo-item">1</div>
          </SwiperItem>
          <SwiperItem>
            <div className="demo-item">2</div>
          </SwiperItem>
          <SwiperItem>
            <div className="demo-item">3</div>
          </SwiperItem>
        </Swiper>
      </>
    );

    render(<TestView />);
    // 获取 element
    expect(document.querySelector('.t-swiper__arrow-left')).not.toBeNull();
    expect(document.querySelector('.t-swiper__arrow-right')).not.toBeNull();
    expect(document.querySelector('.t-swiper__container')).not.toBeNull();

    fireEvent.click(document.querySelector('.t-swiper__arrow-right'));
    expect(document.querySelector('.t-swiper__container').getAttribute('style')).toBe(
      'transform: translate3d(-100%, 0px, 0px); transition: transform 0.3s ease;',
    );
    fireEvent.click(document.querySelector('.t-swiper__arrow-left'));
  });
});
