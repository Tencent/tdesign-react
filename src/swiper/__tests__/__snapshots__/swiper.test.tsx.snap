// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block--swiper-new"
  >
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--inside"
      >
        <div
          class="t-swiper__content"
        >
          <div
            class="t-swiper__container"
            style="transform: translate3d(-0%, 0px, 0px);"
          >
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                6
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <ul
          class="t-swiper__navigation t-swiper__navigation-bars"
        >
          <li
            class="t-swiper__navigation-item t-is-active"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
        </ul>
        <div
          class="t-swiper__arrow t-swiper__arrow--default"
        >
          <div
            class="t-swiper__arrow-left"
          >
            <svg
              class="t-icon t-icon-chevron-left"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M9.54 3.54l.92.92L6.92 8l3.54 3.54-.92.92L5.08 8l4.46-4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-swiper__arrow-right"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`card.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block--swiper-new"
  >
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--inside"
      >
        <div
          class="t-swiper__content t-swiper-card"
        >
          <div
            class="t-swiper__container"
            style="height: 280px;"
          >
            <div
              class="t-swiper__container__item t-swiper__card t-is-active"
              style="transform: translateX(NaNpx) scale(1); transition: transform 0.3s ease; z-index: 2;"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__card"
              style="transform: translateX(NaNpx) scale(0.6325301204819277); transition: transform 0.3s ease; z-index: 1;"
            >
              <div
                class="demo-item"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__card"
              style="transform: translateX(NaNpx) scale(0.6325301204819277); transition: transform 0.3s ease; z-index: 0;"
            >
              <div
                class="demo-item"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__card"
              style="transform: translateX(NaNpx) scale(0.6325301204819277); transition: transform 0.3s ease; z-index: 0;"
            >
              <div
                class="demo-item"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__card"
              style="transform: translateX(NaNpx) scale(0.6325301204819277); transition: transform 0.3s ease; z-index: 0;"
            >
              <div
                class="demo-item"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__card"
              style="transform: translateX(NaNpx) scale(0.6325301204819277); transition: transform 0.3s ease; z-index: 1;"
            >
              <div
                class="demo-item"
              >
                6
              </div>
            </div>
          </div>
        </div>
        <ul
          class="t-swiper__navigation t-swiper__navigation-bars"
        >
          <li
            class="t-swiper__navigation-item t-is-active"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
        </ul>
        <div
          class="t-swiper__arrow t-swiper__arrow--default"
        >
          <div
            class="t-swiper__arrow-left"
          >
            <svg
              class="t-icon t-icon-chevron-left"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M9.54 3.54l.92.92L6.92 8l3.54 3.54-.92.92L5.08 8l4.46-4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-swiper__arrow-right"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`current.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block--swiper-new"
  >
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--inside"
      >
        <div
          class="t-swiper__content"
        >
          <div
            class="t-swiper__container"
            style="transform: translate3d(-0%, 0px, 0px); transition: transform 0.3s ease;"
          >
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                6
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <ul
          class="t-swiper__navigation t-swiper__navigation-bars"
        >
          <li
            class="t-swiper__navigation-item t-is-active"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
        </ul>
      </div>
    </div>
    <div
      class="tdesign-demo-block-row"
    >
      <button
        class="t-button t-button--theme-primary t-button--variant-base t-size-s"
        type="button"
      >
        <span
          class="t-button__text"
        >
          跳转到第 2 项
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`fade.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block--swiper-new"
  >
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--inside"
      >
        <div
          class="t-swiper__content t-swiper-fade"
        >
          <div
            class="t-swiper__container"
            style="height: 280px;"
          >
            <div
              class="t-swiper__container__item t-swiper__fade"
              style="opacity: 1;"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__fade"
              style="opacity: 0;"
            >
              <div
                class="demo-item"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__fade"
              style="opacity: 0;"
            >
              <div
                class="demo-item"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__fade"
              style="opacity: 0;"
            >
              <div
                class="demo-item"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__fade"
              style="opacity: 0;"
            >
              <div
                class="demo-item"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__fade"
              style="opacity: 0;"
            >
              <div
                class="demo-item"
              >
                6
              </div>
            </div>
            <div
              class="t-swiper__container__item t-swiper__fade"
              style="opacity: 0;"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <ul
          class="t-swiper__navigation t-swiper__navigation-bars"
        >
          <li
            class="t-swiper__navigation-item t-is-active"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
        </ul>
        <div
          class="t-swiper__arrow t-swiper__arrow--default"
        >
          <div
            class="t-swiper__arrow-left"
          >
            <svg
              class="t-icon t-icon-chevron-left"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M9.54 3.54l.92.92L6.92 8l3.54 3.54-.92.92L5.08 8l4.46-4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-swiper__arrow-right"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`fraction.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block--swiper-new"
  >
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--inside"
      >
        <div
          class="t-swiper__content"
        >
          <div
            class="t-swiper__container"
            style="transform: translate3d(-0%, 0px, 0px);"
          >
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                6
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <div
          class="t-swiper__navigation t-swiper__navigation--fraction"
        >
          <div
            class="t-swiper__arrow"
          >
            <div
              class="t-swiper__arrow-left"
            >
              <svg
                class="t-icon t-icon-chevron-left"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M9.54 3.54l.92.92L6.92 8l3.54 3.54-.92.92L5.08 8l4.46-4.46z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <div
              class="t-swiper__navigation-text-fraction"
            >
              1/6
            </div>
            <div
              class="t-swiper__arrow-right"
            >
              <svg
                class="t-icon t-icon-chevron-right"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`placement.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block--swiper-new"
  >
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--outside"
      >
        <div
          class="t-swiper__content"
        >
          <div
            class="t-swiper__container"
            style="transform: translate3d(-0%, 0px, 0px);"
          >
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                6
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <ul
          class="t-swiper__navigation t-swiper__navigation-bars"
        >
          <li
            class="t-swiper__navigation-item t-is-active"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
        </ul>
        <div
          class="t-swiper__arrow t-swiper__arrow--default"
        >
          <div
            class="t-swiper__arrow-left"
          >
            <svg
              class="t-icon t-icon-chevron-left"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M9.54 3.54l.92.92L6.92 8l3.54 3.54-.92.92L5.08 8l4.46-4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-swiper__arrow-right"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`size.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block--swiper-new"
    style="width: 1000px;"
  >
    <h3>
      large
    </h3>
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--inside t-swiper--large"
      >
        <div
          class="t-swiper__content"
        >
          <div
            class="t-swiper__container"
            style="transform: translate3d(-0%, 0px, 0px);"
          >
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-large"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-large"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-large"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-large"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-large"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-large"
              >
                6
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-large"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <ul
          class="t-swiper__navigation t-swiper__navigation-bars"
        >
          <li
            class="t-swiper__navigation-item t-is-active"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
        </ul>
        <div
          class="t-swiper__arrow t-swiper__arrow--default"
        >
          <div
            class="t-swiper__arrow-left"
          >
            <svg
              class="t-icon t-icon-chevron-left"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M9.54 3.54l.92.92L6.92 8l3.54 3.54-.92.92L5.08 8l4.46-4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-swiper__arrow-right"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="tdesign-demo-block--swiper-new"
    style="width: 500px; margin-top: 20px;"
  >
    <h3>
      small
    </h3>
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--inside t-swiper--small"
      >
        <div
          class="t-swiper__content"
        >
          <div
            class="t-swiper__container"
            style="transform: translate3d(-0%, 0px, 0px);"
          >
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-small"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-small"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-small"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-small"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-small"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-small"
              >
                6
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item demo-small"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <ul
          class="t-swiper__navigation t-swiper__navigation-bars"
        >
          <li
            class="t-swiper__navigation-item t-is-active"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
        </ul>
        <div
          class="t-swiper__arrow t-swiper__arrow--default"
        >
          <div
            class="t-swiper__arrow-left"
          >
            <svg
              class="t-icon t-icon-chevron-left"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M9.54 3.54l.92.92L6.92 8l3.54 3.54-.92.92L5.08 8l4.46-4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-swiper__arrow-right"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`vertical.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block--swiper-new"
  >
    <div
      class="t-swiper"
    >
      <div
        class="t-swiper__wrap t-swiper--inside t-swiper--vertical"
      >
        <div
          class="t-swiper__content"
        >
          <div
            class="t-swiper__container"
            style="height: 280px; transform: translate3d(0, -0%, 0px);"
          >
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                2
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                3
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                4
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                5
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                6
              </div>
            </div>
            <div
              class="t-swiper__container__item"
            >
              <div
                class="demo-item"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <ul
          class="t-swiper__navigation t-swiper__navigation-bars"
        >
          <li
            class="t-swiper__navigation-item t-is-active"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
          <li
            class="t-swiper__navigation-item"
          >
            <span />
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
